src/smc-hub/client.coffee

```diff
@@ -1162,20 +1162,123 @@ class exports.Client extends EventEmitter
 
     mesg_invite_collaborator: (mesg) =>
         @touch()
+        #dbg = @dbg('mesg_invite_collaborator')
+        #dbg("mesg: #{misc.to_json(mesg)}")
         @get_project mesg, 'write', (err, project) =>
             if err
                 return
+            locals =
+                email_address : undefined
+                done          : false
+
             # SECURITY NOTE: mesg.project_id is valid and the client has write access, since otherwise,
             # the @get_project function above wouldn't have returned without err...
-            @database.add_user_to_project
-                project_id : mesg.project_id
-                account_id : mesg.account_id
-                group      : 'collaborator'  # in future will be "invite_collaborator", once implemented
-                cb         : (err) =>
-                    if err
-                        @error_to_client(id:mesg.id, error:err)
+            async.series([
+                (cb) =>
+                    @database.add_user_to_project
+                        project_id   : mesg.project_id
+                        account_id   : mesg.account_id
+                        group        : 'collaborator'  # in future will be "invite_collaborator", once implemented
+                        cb           : cb
+
+                (cb) =>
+                    {one_result} = require('./postgres')
+                    @database._query
+                        query : "SELECT email_address FROM accounts"
+                        where : "account_id = $::UUID" : mesg.account_id
+                        cb    : one_result 'email_address', (err, x) =>
+                            locals.email_address = x
+                            cb(err)
+
+                (cb) =>
+                    if (not locals.email_address)
+                        cb(); return
+
+                    # INFO: for testing this, you have to reset the invite field each time you sent yourself an invitation
+                    # in psql: UPDATE projects SET invite = NULL WHERE project_id = '<UUID of your cc-in-cc dev project>';
+                    @database.when_sent_project_invite
+                        project_id : mesg.project_id
+                        to         : locals.email_address
+                        cb         : (err, when_sent) =>
+                            if err
+                                cb(err)
+                            else if when_sent - 0 >= misc.days_ago(7) - 0 # successfully sent < one week ago -- don't again
```

Comment:
I think Date objects compare fine when using >= -- no need to convert lhs and rhs by subtracting 0.